src/parser/jobs/mch/modules/Heat.js

```diff
@@ -7,244 +7,71 @@ import {ActionLink} from 'components/ui/DbLink'
+			this._overheatWindows.history.push(this._overheatWindows.current)
+			this._overheatWindows.current = null
 		}
 	}
 
 	_onCast(event) {
-		// This is a bit goofy but here's the reasoning: -Any- action while channeling FT will disrupt it, and we need to add the heat
-		// as soon as that happens (before the interrupting action is handled). The function will add heat if we have an application
-		// timestamp stored, and it clears the flag out in the process, so we won't need to worry about remove getting fired twice.
-		this._onRemoveFlamethrower(event)
-
-		const abilityId = event.ability.guid
+		if (this._overheatWindows.current === null) {
+			return
+		}
 
-		switch (this._barrelState) {
-		case BARREL_STATE.NORMAL:
-			if (abilityId === ACTIONS.GAUSS_BARREL.id) {
-				this._setBarrelState(BARREL_STATE.GAUSS)
-			}
-			break
-		case BARREL_STATE.GAUSS:
-			if (abilityId === ACTIONS.GAUSS_BARREL.id) {
-				this._fixNullAssumption()
-			} else if (abilityId === ACTIONS.REMOVE_BARREL.id) {
-				// y tho
-				this._heat = 0
-				this._setBarrelState(BARREL_STATE.NORMAL)
-			} else if (HEAT_BUILDERS.hasOwnProperty(abilityId) && !this.ammo.ammoSpent) {
-				this._addHeat(HEAT_BUILDERS[abilityId])
-			} else if (abilityId === ACTIONS.COOLDOWN.id) {
-				this._heat = Math.max(this._heat - COOLDOWN_HEAT_REDUCTION, 0) // Floor at 0, just to be safe
-				if (this._heat < HEATED_POINT) {
-					this._badCooldowns++
-				}
-			} else if (abilityId === ACTIONS.BARREL_STABILIZER.id) {
-				this._heat = HEATED_POINT
-			}
-			break
-		case BARREL_STATE.OVERHEATED:
-			if (abilityId === ACTIONS.GAUSS_BARREL.id) {
-				this._fixNullAssumption()
-			} else if (event.timestamp - this._lastBarrelStateChange < OVERHEAT_DURATION_MILLIS) {
-				this._overheatWindows.current.casts.push(event)
-				if (this.ammo.ammoSpent) {
-					// Don't count typically bad ammo expenditures if they happen during OH
-					this.ammo.negateBadAmmoUse(abilityId)
-				}
-			} else {
-				this._overheatWindows.current.gcdCount = this._overheatWindows.current.casts.filter(cast => {
-					const action = getDataBy(ACTIONS, 'id', cast.ability.guid)
-					return action && action.onGcd
-				}).length
-				this._overheatWindows.history.push(this._overheatWindows.current)
-				this._heat = 0
-				this._setBarrelState(BARREL_STATE.COOLING, OVERHEAT_DURATION_MILLIS)
-			}
-			break
-		case BARREL_STATE.COOLING:
-			if (event.timestamp - this._lastBarrelStateChange >= COOLING_DURATION_MILLIS - COOLING_FUZZ_MILLIS) {
-				this._setBarrelState(BARREL_STATE.NORMAL, COOLING_DURATION_MILLIS)
-				if (abilityId === ACTIONS.GAUSS_BARREL.id) {
-					this._setBarrelState(BARREL_STATE.GAUSS)
-				}
-			} else if (abilityId === ACTIONS.COOLDOWN.id) {
-				this._lockoutCooldowns++
-			} else if (abilityId === ACTIONS.GAUSS_BARREL.id) {
-				this._fixNullAssumption()
-			}
-			break
+		if (event.timestamp > (this._overheatWindows.current.start + OVERHEAT_DURATION_MILLIS)) {
+			this._finishOverheatWindow()
```

Comment:
I'm guessing there's no OH status to track?

Reply:
Yeh, OH is still gauge only. It just shows an 8 ticking down on your heat bar when you pop Hypercharge.